### PR TITLE
SDK: Add support for Azure client secret auth

### PR DESCRIPTION
**Pull request by fjakobs**
## Changes
SDK: Add support for Azure client secret auth

Ported from https://github.com/databricks/databricks-sdk-go/blob/main/config/auth_azure_client_secret.go
 
## Tests
Manually tested with Azure SP credentials.

### CODE REVIEW

File: packages/databricks-sdk-js/src/AccountClient.ts [hunk 1]
```diff
@@ -282,12 +282,6 @@ export class AccountClient {
         }
 
         this.config = config as Config;
-
-        this.config.ensureResolved();
-        if (!this.config.accountId || !this.config.isAccountClient()) {
-            throw new Error("invalid Databricks Account configuration");
-        }
```
Comment (kartikgupta-db) [inline packages/databricks-sdk-js/src/AccountClient.ts:289]: Why did you move it out? I don't understand the use of moving it out.

File: packages/databricks-sdk-js/examples/workspace.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import {Config, WorkspaceClient} from "..";
+import {Config, WorkspaceClient} from "../src";
```
Comment (kartikgupta-db) [inline packages/databricks-sdk-js/examples/workspace.ts:1]: Let's make examples it's own package. Easier for people to understand imports that way and easier to get started.

File: packages/databricks-sdk-js/src/AccountClient.ts (hunk 1 above)
Comment (fjakobs) [reply packages/databricks-sdk-js/src/AccountClient.ts:289]: `ensureResolved` is async but since this is the constructor we can't await for this. 
This means that the check below never actually worked. I moved the check into the actual calls.

File: packages/databricks-sdk-js/package.json [hunk 3]
```diff
@@ -35,6 +35,7 @@
         "ini": "^4.1.1",
         "inversify": "^6.0.1",
         "node-fetch-commonjs": "^3.2.4",
+        "openid-client": "^5.4.2",
```
Comment (nfx) [inline packages/databricks-sdk-js/package.json:38]: why do we need it?

File: packages/databricks-sdk-js/src/.codegen/AccountClient.ts.tmpl [hunk 4]
```diff
@@ -27,12 +27,6 @@ export class AccountClient {
         }
 
         this.config = config as Config;
-
-        this.config.ensureResolved();
```
Comment (nfx) [inline packages/databricks-sdk-js/src/.codegen/AccountClient.ts.tmpl:31]: why do we need to move it out of constructor?

File: packages/databricks-sdk-js/src/config/AzureClientSecretCredentials.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    getAzureLoginAppId,
+} from "./Azure";
+import {
+    AuthType,
+    Config,
+    CredentialProvider,
+    RequestVisitor,
+    Headers,
+} from "./Config";
+import {Issuer, TokenSet} from "openid-client";
```
Comment (nfx) [inline packages/databricks-sdk-js/src/config/AzureClientSecretCredentials.ts:14]: please re-implement it without external dependencies as in Go / Python / Java SDKs - it's just resolving a couple of URLs.

File: packages/databricks-sdk-js/src/config/AzureClientSecretCredentials.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+                    resource: [env.serviceManagementEndpoint],
+                });
+            }
+
+            if (config.azureResourceId) {
+                headers["X-Databricks-Azure-Workspace-Resource-Id"] =
+                    config.azureResourceId;
+            }
+            headers["Authorization"] = `Bearer ${innerToken.access_token}`;
+            headers["X-Databricks-Azure-SP-Management-Token"] =
```
Comment (nfx) [inline packages/databricks-sdk-js/src/config/AzureClientSecretCredentials.ts:62]: shouldn't it get a surroudning `if`?

**Review by nfx — CHANGES_REQUESTED** [review]

**Review by kartikgupta-db — APPROVED** [review]

File: packages/databricks-sdk-js/src/AccountClient.ts (hunk 1 above)
Comment (kartikgupta-db) [reply packages/databricks-sdk-js/src/AccountClient.ts:289]: Makes sense. Thanks!

File: packages/databricks-sdk-js/src/config/AzureClientSecretCredentials.ts (hunk 6 above)
Comment (fjakobs) [reply packages/databricks-sdk-js/src/config/AzureClientSecretCredentials.ts:68]: what should be in that `if`?

File: packages/databricks-sdk-js/package.json (hunk 3 above)
Comment (fjakobs) [reply packages/databricks-sdk-js/package.json]: I removed the dependency

**Review by kartikgupta-db — APPROVED** [review]
